### ISSUE TITLE
Segfault in GetMetadata

### ISSUE THREAD

**Issue by mtibben** [99designs/aws-vault#573]
From @gws in https://github.com/99designs/aws-vault/pull/569#issuecomment-622044667

#569 (possibly) breaks for me with the `pass` backend on Linux. Every operation (this one was just `list`, but it happens on `exec` and `rm` too that I tested) results in this panic, but only **after starting a session**:

```
$ aws-vault --debug list
2020/04/30 12:02:56 aws-vault v5.5.0-beta2
2020/04/30 12:02:56 [keyring] Considering backends: [pass]
2020/04/30 12:02:56 Loading config file /home/me/.aws/config
2020/04/30 12:02:56 Parsing config file /home/me/.aws/config
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x9a8d1f]

goroutine 1 [running]:
github.com/99designs/aws-vault/v5/vault.(*SessionKeyring).GetMetadata(0xc000161b80, 0xc0001386fd, 0x13, 0xc00002373e, 0x2, 0xc000026c60, 0x24, 0x0, 0x0, 0x0, ...)
        github.com/99designs/aws-vault/v5/vault/sessionkeyring.go:172 +0x1af
github.com/99designs/aws-vault/v5/vault.(*SessionKeyring).GetAllMetadata(0xc000161b80, 0xc0000712c0, 0x4, 0x4, 0x0, 0x0)
        github.com/99designs/aws-vault/v5/vault/sessionkeyring.go:190 +0x143
github.com/99designs/aws-vault/v5/cli.LsCommand(0xc000000000, 0xc00000dca0, 0xd57da0, 0xc0001fb9b0, 0x0, 0x0)
        github.com/99designs/aws-vault/v5/cli/ls.go:66 +0xdfe
github.com/99designs/aws-vault/v5/cli.ConfigureListCommand.func1(0xc0001f19e0, 0x40a5b3, 0xacec20)
        github.com/99designs/aws-vault/v5/cli/ls.go:45 +0xd0
gopkg.in/alecthomas/kingpin%2ev2.(*actionMixin).applyActions(0xc00021a0d8, 0xc0001f19e0, 0x0, 0x0)
        gopkg.in/alecthomas/kingpin.v2@v2.2.6/actions.go:28 +0x6d
gopkg.in/alecthomas/kingpin%2ev2.(*Application).applyActions(0xc000218690, 0xc0001f19e0, 0x0, 0x0)
        gopkg.in/alecthomas/kingpin.v2@v2.2.6/app.go:557 +0xdc
gopkg.in/alecthomas/kingpin%2ev2.(*Application).execute(0xc000218690, 0xc0001f19e0, 0xc000055a40, 0x1, 0x1, 0x0, 0x0, 0x0, 0xae11a0)
        gopkg.in/alecthomas/kingpin.v2@v2.2.6/app.go:390 +0x8f
gopkg.in/alecthomas/kingpin%2ev2.(*Application).Parse(0xc000218690, 0xc00001e0a0, 0x2, 0x2, 0xc000218690, 0x405aff, 0xc00002a118, 0x0)
        gopkg.in/alecthomas/kingpin.v2@v2.2.6/app.go:222 +0x1fe
main.main()
        github.com/99designs/aws-vault/v5/main.go:26 +0x184

```

I was using 5.5.0-beta1 successfully in order to test out the YubiKey support which worked well. I can work around the panic by removing the session file from the backend manually and at least start up a session that way, but it's painful.

I'm using MFA via YubiKey which is what led me to believe this change is the culprit.

**Comment by mtibben** [comment]
Related to https://github.com/99designs/keyring/pull/70

**Comment by mtibben** [comment]
It looks like the keyring's GetMetadata is behaving differently for different backends. Specifically the Item is nil, and some backends fill the Item data and some don't. That's a problem, we can't rely on this API if it's not functional for some backends. 

Any thoughts on if we can get this API behaving consistently between backends @philpennock?

**Comment by philpennock** [comment]
Hrm, I don't know what happened to the godoc, but I remember explicitly documenting that GetMetadata would return non-nil if data could be retrieved, but nil if it couldn't.  For something which is "get metadata without prompting for passwords", if data just isn't available, there's nothing to return.  Callers have to check for a pointer being non-nil.

That said, I would imagine that `pass` could stat the file on disk to get a timestamp and could return non-nil easily enough.  I think this was one of the backends added between when I wrote the original PR and when it was merged, and someone added a stub to get things compiling again?
